.github/workflows/ci.yml

```diff
@@ -47,7 +47,7 @@ jobs:
         uses: actions/cache@v3.3.2
         with:
           path: ${{ steps.setup-haskell.outputs.cabal-store }}
-          key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+          key: ${{ runner.os }}-${{ matrix.ghc }}
```

Comment:
Ah, sorry, I think you want some part in the cache key that changes when a dependency gets updated.  Otherwise, your cache gets stale, keeping builds of old versions of the dependencies.
The trick is really to use `restore-keys`.
Please have a look at: https://github.com/agda/agda/blob/7810340af033c65f3ac20d0b09b85cc1a5812fb1/src/github/workflows/cabal.yml#L181-L182
```
key:          cabal.yml-${{ runner.os }}-ghc-${{ env.GHC_VER }}-cabal-${{ env.CABAL_VER }}-${{ hashFiles('**/plan.json') }}
restore-keys: cabal.yml-${{ runner.os }}-ghc-${{ env.GHC_VER }}-cabal-${{ env.CABAL_VER }}-
```
I use the hash of `dist-newstyle/cache/plan.json` for the movable part of the cache key.  This one is better than the freeze file because it does not contain a timestamp.

Reply:
aaah yes I see :)